### PR TITLE
Wait a while for OpenShift API server to apply last changes

### PR DESCRIPTION
**Pull request by danpawlik**
In some corner cases, when infrastructure might be overloaded, the OpenShift API is not restarted
immediately, but it is done after a while. From basic check on failing job, it happens from up to 90 seconds.
Making condition that would watch on OpenShift API server that has been finally changed state to to terminating then back to active might be a good alternative here, but can also impact future job if such restart is not needed.

Also remove duplicated line from periodic job in zuul config.

### CODE REVIEW

**Comment by danpawlik** [comment]
hm, still does not work. Hmm

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck
it is again same situation - it always pass, where the task even don't take 10 seconds, and after a while it would raise an error. What a mistery

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck
again success.

**Comment by danpawlik** [comment]
recheck
again success.

**Comment by danpawlik** [comment]
recheck
again success.

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck
again success.

File: build/stf-run-ci/tasks/main.yml [hunk 1]
```diff
@@ -40,23 +40,15 @@
...
-  kubernetes.core.k8s_info:
-    api_version: config.openshift.io/v1
-    kind: ClusterOperator
-    name: openshift-apiserver
-  register: _co_status
-  until: >
-    (_co_status.resources | length > 0) and
-    (_co_status.resources[0].status.conditions | selectattr('type', 'equalto', 'Available') | map(attribute='status') | first == 'True')
-  retries: 60
-  delay: 10
```
Comment (vyzigold) [inline build/stf-run-ci/tasks/main.yml:58]: Why is this task getting deleted? Is it basically a duplicate of the task above?

File: build/stf-run-ci/tasks/main.yml [hunk 2]
```diff
@@ -40,23 +40,15 @@
     - __deploy_from_index_enabled | bool
     - __deploy_from_catalog | bool
 
+- name: Wait a minute to apply
+  ansible.builtin.pause:
+    minutes: 2
```
Comment (vyzigold) [inline build/stf-run-ci/tasks/main.yml]: We generally try to avoid straight up waiting for a hardcoded period of time and instead try to have something that can be watched with a retry. But do I understand it correctly, that the OpenShift API may or may not restart and it may take some time for that restart to begin. So because of that it's hard to find anything that we can watch? If that's the case, then the pause in fine by me.

File: build/stf-run-ci/tasks/main.yml (hunk 1 above)
Comment (danpawlik) [reply build/stf-run-ci/tasks/main.yml:58]: It was just useless. It never catch an issue on 4.18 - always pass. Where after few seconds api was unstable. 
Normally I will do:
```
oc adm wait-for-stable-cluster --minimum-stable-period=120s --timeout=30m
```
and it should be fine, but there is a machineconfig issue on CRC images, so it would not pass :|

**Comment by danpawlik** [comment]
@vyzigold making one more test with `oc adm wait-for-stable-cluster --minimum-stable-period=120s --timeout=30m`. Maybe we can bypass mcp

**Comment by vyzigold** [comment]
> @vyzigold making one more test with `oc adm wait-for-stable-cluster --minimum-stable-period=120s --timeout=30m`. Maybe we can bypass mcp

Let's see if it works. If it doesn't, then the previous solution was also OK.

**Comment by danpawlik** [comment]
recheck
That error is very common. Added improvement in https://github.com/openstack-k8s-operators/ci-framework/pull/3943

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
as a result:
```
2026-05-20 12:39:21.205856 | TASK [../build/stf-run-ci : Run script to ensure cluster is stable]
2026-05-20 12:39:21.525665 | controller | Waiting for cluster operators to stabilize (ignoring machine-config)...
2026-05-20 12:39:21.714486 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:31.903842 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:42.077157 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:52.211763 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:02.383294 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:12.561401 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:22.739932 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:32.925054 | controller | All relevant operators stable. Waiting for minimum stable period (120s)...
2026-05-20 12:40:32.925128 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:43.100545 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:53.285605 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:03.469255 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:13.642845 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:23.823114 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:34.032905 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:44.218089 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:54.404317 | controller | Sleeping 10 to met condition...
2026-05-20 12:42:04.560875 | controller | Sleeping 10 to met condition...
2026-05-20 12:42:14.744856 | controller | Sleeping 10 to met condition...
2026-05-20 12:42:24.906071 | controller | Sleeping 10 to met condition...
2026-05-20 12:42:35.060755 | controller | Cluster is stable (machine-config ignored).
2026-05-20 12:42:35.536769 | controller | ok: Runtime: 0:03:13.540740
```
another:
```
2026-05-20 12:39:23.526977 | TASK [../build/stf-run-ci : Run script to ensure cluster is stable]
2026-05-20 12:39:23.851741 | controller | Waiting for cluster operators to stabilize (ignoring machine-config)...
2026-05-20 12:39:24.049565 | controller | All relevant operators stable. Waiting for minimum stable period (120s)...
2026-05-20 12:39:24.049661 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:34.260488 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:44.426040 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:54.641239 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:04.814205 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:14.994920 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:25.157221 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:35.372561 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:45.514810 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:55.703972 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:05.845144 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:16.050207 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:26.261241 | controller | Cluster is stable (machine-config ignored).
```
one more:
```
2026-05-20 12:38:08.944002 | TASK [../build/stf-run-ci : Run script to ensure cluster is stable]
2026-05-20 12:38:09.246889 | controller | Waiting for cluster operators to stabilize (ignoring machine-config)...
2026-05-20 12:38:09.429234 | controller | Sleeping 10 to met condition...
2026-05-20 12:38:19.572460 | controller | Sleeping 10 to met condition...
2026-05-20 12:38:29.741912 | controller | Sleeping 10 to met condition...
2026-05-20 12:38:39.913815 | controller | Sleeping 10 to met condition...
2026-05-20 12:38:50.076907 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:00.276999 | controller | All relevant operators stable. Waiting for minimum stable period (120s)...
2026-05-20 12:39:00.277086 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:10.455838 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:20.618642 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:30.816083 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:40.986285 | controller | Sleeping 10 to met condition...
2026-05-20 12:39:51.177901 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:01.378842 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:11.566615 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:21.747638 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:31.942215 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:42.136706 | controller | Sleeping 10 to met condition...
2026-05-20 12:40:52.343688 | controller | Sleeping 10 to met condition...
2026-05-20 12:41:02.540963 | controller | Cluster is stable (machine-config ignored).
```

It is soo random. But I guess it is stable enough right now. Wdyt @vyzigold ?

File: .zuul.yaml [hunk 3]
```diff
@@ -243,4 +243,3 @@
         - stf-crc-ocp_420-local_build-index_deploy
         - stf-crc-ocp_418-nightly_bundles-index_deploy
         - stf-crc-ocp_418-nightly_bundles-index_deploy
```
Comment (vyzigold) [inline .zuul.yaml]: since you're mentioning it in the PR description, you should remove the duplicate :D

File: build/stf-run-ci/templates/check-cluster-co.sh.j2 [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            exit 0
+        fi
+    else
+        if ((stable_since != 0)); then
+            echo "Operators became unstable again: ${unstable[*]}"
+        fi
+        stable_since=0
+    fi
+
+    echo "Sleeping $POLL to met condition..."
```
Comment (vyzigold) [inline build/stf-run-ci/templates/check-cluster-co.sh.j2]:
```suggestion
    echo "Sleeping $POLL to meet condition..."
```

**Comment by vyzigold** [comment]
I like this approach better than the sleep there was before. You'll need to either delete the duplicate line in .zuul.yaml or at least remove that part of the PR description. After I can approve.

**Comment by vkmc** [comment]
Thanks for your work on this Daniel!

For what I understand, it is a specific issue in the CRC image for OCP 4.18 and it is not reproducible in the 2 other CIs (OCP 4.16 and OCP 4.20).

I'm not convinced with the approach of adding a script to check on cluster stability instead of using the already existing tooling.

If this is more reliable than the existing tool, though, I'm ok to merge it.

**Comment by danpawlik** [comment]
@vkmc so far, for CRC there is no additional tool that can verify if the cluster is stable right after rollback. I only see that issue on 4.18, that's why the script was done. It might be helpful for other job, where similar use case was done.

**Review by vkmc — APPROVED** [review]
LGTM

**Review by vyzigold — APPROVED** [review]
